.github/workflows/test.yml

```diff
@@ -7,6 +7,75 @@ on:
   pull_request:
 
 jobs:
+  check_submodule_token:
+    name: Check if the secret exists or not.
+    runs-on: ubuntu-latest
+    outputs:
+      has_token: ${{ steps.secret-check.outputs.has_token }}
+    steps:
+      - name: Check if secret exists
+        id: secret-check
+        run: |
+          if [[ -n "${{ secrets.SUBMODULES_TOKEN }}" ]]; then
+            echo "::set-output name=has_token::true"
+          else
+            echo "::set-output name=has_token::false"
+          fi
+
+  trigger-e2e-ee:
+    name: RUN E2E-EE TESTS
+    runs-on: ubuntu-latest
+    timeout-minutes: 80
+    permissions:
+      contents: read
+      packages: write
+      deployments: write
+      id-token: write
+    needs: [check_submodule_token]
+    steps:
+      # checkout with submodules if token is provided
+      - uses: actions/checkout@v3
+        if: ${{ needs.check_submodule_token.outputs.has_token == 'true' }}
+        with:
+          submodules: true
+          token: ${{ secrets.SUBMODULES_TOKEN }}
+      # else checkout without submodules if the token is not provided
+      - uses: actions/checkout@v3
+        if: ${{ needs.check_submodule_token.outputs.has_token != 'true' }}
+      - uses: ./.github/actions/setup-project
+        with:
+          submodules: true
+      - uses: ./.github/actions/setup-redis-cluster
+      - uses: mansagroup/nrwl-nx-action@v3
+        with:
+          targets: lint
+          projects: "@novu/api"
+
+      - uses: ./.github/actions/start-localstack
+
+      - uses: ./.github/actions/run-worker
+        with:
+          launch_darkly_sdk_key: ${{ secrets.LAUNCH_DARKLY_SDK_KEY }}
+
+        # Runs a single command using the runners shell
+      - name: Build API
+        run: CI='' pnpm build:api
+
+      - name: Build Submodules
+        run: CI='' pnpm build:ee-auth
```

Comment:
maybe we should have a command that will run the script where we will list and build all ee deps?

Reply:
For now, we need to build the ee-auth module, am i missing something? or did you have something else in mind?
@LetItRock

Reply:
I thought about having just one command to build all the ee packages ;) rn we are building only this one, but it will change when we have translation management ready... and then we would need to update this pipeline

Reply:
I see, we could.
Here i was afraid to build redundant libs so the runtime would increase, but we could build it all for now so it will be more simple and then optimize if needed.